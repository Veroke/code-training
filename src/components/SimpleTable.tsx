import { defineComponent } from "vue";
import { type TableProps, tableProps, ListItemType } from "./types";
import axios from 'axios'
import { ref } from 'vue-demi'
import './index.less'

export default defineComponent({
  name: "SimpleTable",
  props: tableProps,
  setup(props: TableProps, { attrs, emit, slots }) {
    let isSort = false
    const headList = [
      {
        prop: 'name',
        label: '姓名',
      },
      {
        prop: 'sex',
        label: '性别'
      },
      {
        prop: 'age',
        label: '年龄',
        isSort: true
      },
      {
        prop: 'address',
        label: '地址'
      }
    ]

    let bodyList = ref<ListItemType[]>([])
    function headClick(prop) {
      isSort = !isSort
      sort(prop, isSort ? '' : 'asc')
    }

    function getList() {
      axios('user/list').then((res) => {
        bodyList.value = res.data.list
        console.log('bodyList', bodyList.value);

      })

    }

    getList()

    // 排序
    function sort(type, order) {
      let table = document.getElementById("tbody");
      let trArr = table?.children;
      let intType;
      // TODO 目前只做了年龄的排序，后续做到自定义排序
      if (type == "age") {
        intType = 2;
      }
      let temp
      if (order == "asc") {
        for (let i = 0; i < trArr.length - 1; i++) {
          for (let j = 0; j < trArr.length - i - 1; j++) {
            if ((trArr[j].children[intType].innerHTML - 0) > (trArr[j + 1].children[intType].innerHTML - 0)) {
              temp = trArr[j].innerHTML;
              trArr[j].innerHTML = trArr[j + 1].innerHTML;
              trArr[j + 1].innerHTML = temp;
            }
          }
        }
      } else {
        for (let i = 0; i < trArr.length - 1; i++) {
          for (let j = 0; j < trArr.length - i - 1; j++) {
            if ((trArr[j].children[intType].innerHTML - 0) < (trArr[j + 1].children[intType].innerHTML - 0)) {
              temp = trArr[j].innerHTML;
              trArr[j].innerHTML = trArr[j + 1].innerHTML;
              trArr[j + 1].innerHTML = temp;
            }
          }
        }
      }
    }
    return () => {
      return (
        <table class="table">
          <thead >
            <tr class="head-tr">
              {
                headList.map(item => {
                  return <th class="head-th">
                    {item.label}
                    {
                      item.isSort && <i onClick={() => { headClick(item.prop) }}>^</i>
                    }
                  </th>
                })
              }
            </tr>
          </thead>
          <tbody id="tbody">
            {
              bodyList.value.map(item => {
                return <tr class="body-tr">
                  <td class="body-td">{item.name}</td>
                  <td class="body-td">{item.sex}</td>
                  <td class="body-td">{item.age}</td>
                  <td class="body-td">{item.address}</td>
                </tr>
              })
            }
          </tbody>
        </table>
      );
    };
  },
});
