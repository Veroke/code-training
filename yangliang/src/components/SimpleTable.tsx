import { defineComponent, onErrorCaptured, computed, reactive, onMounted } from "vue";
import TableHeader from "./table/header/tableHeader/tableHeader";
import TableBody from "./table/body/tableBody/TableBody"
import Pagination from "./pagination/Pagination";
import { Column } from "./types";
import { addLog } from '../log/log';
import { useSorter } from '../hook/useSorter';
// import log4js from "log4js";
// log4js.configure({
//   appenders: { cheese: { type: "file", filename: "cheese.log" } },
//   categories: { default: { appenders: ["cheese"], level: "error" } }
// });

// const logger = log4js.getLogger();

type TableProps = {
    columns: Column[];
    dataSource: any[];
    pageSize: number;
    currentPage: number;
    total: number;
    showQuickJumper: boolean;
    paginationChange: () => {};
}


export default defineComponent({
  name: "SimpleTable",
  props: [
    'columns',
    'dataSource',
    'pageSize',
    'currentPage',
    'total',
    'showQuickJumper',
    'paginationChange'
  ],
  setup(props: TableProps, { attrs, emit, slots }) {
    
    onMounted(()=>{
      addLog({
        level: "info",
        message: {
          component: "SimpleTable",
          props,
          msg: 'Some One Use SimpleTable Component'
        }
      })
    })

    onErrorCaptured((err) => {
      addLog({
        level: "error",
        message: {
          component: "SimpleTable",
          props,
          msg: err
        }
      })
    })

    let currentSorter = reactive({
      key: '',
      asc: true
    });

    const getSortedDataSource = computed(() =>{ 
      const currentDataSource = props.dataSource;
      return useSorter(currentDataSource, currentSorter); 
    });
    const sorter = (key: string, asc: boolean) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }

    return () => {
      return (
        <div>
          <table>
            <TableHeader columns={props.columns} sorter={sorter} />
            <TableBody columns={props.columns}  dataSource={getSortedDataSource.value} />
          </table>
          <Pagination total={props.total}
                      pageSize={props.pageSize}
                      current={props.currentPage}
                      show-quick-jumper={props.showQuickJumper}
                      onChange={props.paginationChange} />
        </div>
      );
    };
  },
});
