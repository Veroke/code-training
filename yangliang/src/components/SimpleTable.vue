<template>
  <table class="table">
    <thead>
      <th v-for="item in tableProps?.columns" :key="item.dataIndex" class="th-td">
        <td>
          {{ item.title }} 
          <img v-if="item.sortOrder" src="../assets/dropup.svg" @click="onSortOrder"> 
          <img v-if="item.sortOrder" src="../assets/dropdown.svg" @click="onSortOrder"> 
        </td>
      </th>
    </thead>
    <tbody>
      <tr v-for="body in getdataSource" :key="body.key">
        <td v-for="head in tableProps?.columns" :key="head.dataIndex" class="tr-td">{{ body[head.key] }}</td>
      </tr>
    </tbody>
  </table>
  <Pagnation :total="total"
             :limit="limit"
             :page="page"
             @page-change="pageChange"
             show-quick-jumper />
</template>

<script setup lang="ts">
import { defineProps, PropType, computed, ref, watch, reactive } from 'vue';
import { TableProp,DataSource } from './types';
import Pagnation from './Pagnation.vue';
const props = defineProps({
    tableProps: Object as PropType<TableProp>
})

const limit = ref(5);

const total = props?.tableProps?.dataSource.length as number;

const page = ref(1);

const dataSource = reactive(props?.tableProps?.dataSource as DataSource[]);

const getdataSource = computed<DataSource[]>({
  get () {
    return dataSource.slice((page.value - 1) * 5, (page.value - 1) * 5 + limit.value);
  },
  set (val) {

  }
})

function pageChange (curPage:number) {  // 页码改变
  if(curPage >= 1 && curPage <= total / limit.value){
    page.value = curPage;
  }
}

function onSortOrder () { // 排序
  dataSource.reverse();
}

</script>

<style scoped>
  .table {
    width: 100%;
  }
  .th-td,.tr-td{
    border: 1px solid #ddd;
  }
 
</style>