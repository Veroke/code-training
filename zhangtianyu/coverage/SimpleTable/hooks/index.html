
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for SimpleTable/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> SimpleTable/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.41% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>313/354</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">87.35% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>76/87</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">94.44% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>17/18</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.41% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>313/354</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="useBaseConfig.ts"><a href="useBaseConfig.ts.html">useBaseConfig.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="useColumns.ts"><a href="useColumns.ts.html">useColumns.ts</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="132" class="abs high">120/132</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="42" class="abs high">35/42</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="132" class="abs high">120/132</td>
	</tr>

<tr>
	<td class="file high" data-value="useData.ts"><a href="useData.ts.html">useData.ts</a></td>
	<td data-value="83.01" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 83%"></div><div class="cover-empty" style="width: 17%"></div></div>
	</td>
	<td data-value="83.01" class="pct high">83.01%</td>
	<td data-value="53" class="abs high">44/53</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="83.01" class="pct high">83.01%</td>
	<td data-value="53" class="abs high">44/53</td>
	</tr>

<tr>
	<td class="file medium" data-value="useFilter.ts"><a href="useFilter.ts.html">useFilter.ts</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="42" class="abs medium">24/42</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="42" class="abs medium">24/42</td>
	</tr>

<tr>
	<td class="file high" data-value="useSortable.ts"><a href="useSortable.ts.html">useSortable.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="88" class="abs high">88/88</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="88" class="abs high">88/88</td>
	</tr>

<tr>
	<td class="file high" data-value="useStyle.ts"><a href="useStyle.ts.html">useStyle.ts</a></td>
	<td data-value="90.47" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.47" class="pct high">90.47%</td>
	<td data-value="21" class="abs high">19/21</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="90.47" class="pct high">90.47%</td>
	<td data-value="21" class="abs high">19/21</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Sat May 28 2022 21:45:23 GMT+0800 (中国标准时间)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    