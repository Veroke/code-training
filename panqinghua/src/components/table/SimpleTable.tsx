import { defineComponent, provide,watch } from "vue";
import { type TableProps, tableProps,ColumnType } from "./types";
import TableHead from "./TableHead"
import TableBody from "./TableBody";
import Pagination from "../pagination";
import { TABLE_PROPS } from "./const"
import { usePageListener } from "../hooks/usePageListener"
import { useTableSortList } from "../hooks/useTableSortList"
export default defineComponent({
  name: "SimpleTable",
  props: tableProps,
  components: {
    Pagination,
    TableBody,
    TableHead
  },
  setup(props: TableProps) {
    let { data, pageSize } = props
    let { current, onPageChange } = usePageListener()
    let { columnItem, updateSortItem } = useTableSortList()
    let tableData = {
      props,
      current,
      columnItem,
    }

    // 提供给body使用
    provide(TABLE_PROPS, tableData)
    return () => {

      return (
        <div>
          <table class="is-bordered is-hoverable is-fullwidth table">
            <TableHead 
              columns={props.columns} 
              onUpdateSortItem={(val) => updateSortItem(val)}
            ></TableHead>
            <TableBody></TableBody>
          </table>
          <Pagination
            total={data.length}
            current={current.value}
            pageSize={pageSize}
            onChange={(val) => onPageChange(val)}
          ></Pagination>
        </div>
      );
    };
  },
});
