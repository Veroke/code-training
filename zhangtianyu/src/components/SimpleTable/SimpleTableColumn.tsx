import { defineComponent } from 'vue'
import { columnProps } from 'components/SimpleTable/type'

export default defineComponent({
    name: 'SimpleTableColumn',
    props: columnProps,
    setup() {
        return () => {
            return <div></div>
        }
    },
})