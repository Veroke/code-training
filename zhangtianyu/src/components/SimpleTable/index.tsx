import { computed, defineComponent, toRefs } from 'vue'
//样式
import './style.less'

import { tableProps } from 'components/SimpleTable/type'
//组件
import TableHeader from './components/TableHeader'
import TableBody from './components/TableBody'
import Pagination from '../Pagination/index.vue'
//hook
import useBaseConfig from './hooks/useBaseConfig'
import useColumns from './hooks/useColumns'
import useSortable from './hooks/useSortable'
import useFilter from './hooks/useFilter'
import useData from './hooks/useData'
import useStyle from './hooks/useStyle'

export default defineComponent({
    name: 'SimpleTable',
    props: tableProps,
    emits: ['sortChange', 'filterChange', 'currentChange'],
    setup(props, { slots, emit }) {

        //收集表格基本配置
        let baseConfig = useBaseConfig(props)

        //收集表格列信息
        let columns = useColumns(baseConfig, slots)

        //处理排序相关信息
        let { activeSorters, onChangeSort } = useSortable(baseConfig, columns, emit)

        //处理筛选相关信息
        let { activeFilters, onFilterChange } = useFilter(baseConfig, columns, emit)

        //处理数据
        let filterData = useData(props, baseConfig, activeSorters, activeFilters)

        //处理表格样式
        let { tableStyle } = useStyle(props)

        let { pageSize, total, pagerCount, currentPage } = toRefs(props)


        return () => {
            return (
                <section class="simple-table" style={tableStyle.value}>
                    <div class="simple-table__header-warp">
                        <TableHeader onChangeSort={onChangeSort} onFilterChange={onFilterChange} columns={columns.value} />
                    </div>
                    <div class="simple-table__body-warp">
                        <TableBody columns={columns.value} data={filterData.value} row-key={props.rowKey} />
                    </div>
                    <div class="simple-table__pagination">
                        <Pagination page-size={pageSize.value} total={total.value} pager-count={pagerCount.value} current-page={currentPage.value} onCurrentChange={(...args: any[]) => emit('currentChange', ...args)} />
                    </div>
                </section>
            )
        }
    },
})