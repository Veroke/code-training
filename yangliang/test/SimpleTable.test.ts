import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import SimpleTable from "../src/components/SimpleTable";
import { columns, mock } from './testData';

test("basic work", () => {
  const wrapper = shallowMount(SimpleTable, {
    props: {
      columns,
      dataSource: mock.dataSource
    },
  });

  expect(wrapper.html()).toMatchSnapshot();

  expect(wrapper.find('ul').findAll('li')[1].classes()).toContain('disabled');
  expect(wrapper.find('ul').findAll('li')[2].classes()).toContain('active');

});


// vi.fn()  expect wrapper.find findAll toContain 
// trigger('click', new Event())  compositionstart
// promise await setProps setValue  