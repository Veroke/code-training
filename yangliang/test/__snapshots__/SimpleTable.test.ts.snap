// Vitest Snapshot v1

exports[`!columns & !dataSource 1`] = `
"<div>
  <table>
    <table-header-stub columns=\\"\\" sorter=\\"(key, asc) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }\\"></table-header-stub>
    <table-body-stub columns=\\"\\" datasource=\\"\\"></table-body-stub>
  </table>
  <pagination-stub></pagination-stub>
</div>"
`;

exports[`!columns & dataSource 1`] = `
"<div>
  <table>
    <table-header-stub columns=\\"\\" sorter=\\"(key, asc) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }\\"></table-header-stub>
    <table-body-stub columns=\\"\\" datasource=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\"></table-body-stub>
  </table>
  <pagination-stub></pagination-stub>
</div>"
`;

exports[`basic work 1`] = `
"<div>
  <table>
    <table-header-stub columns=\\"[object Object],[object Object],[object Object]\\" sorter=\\"(key, asc) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }\\"></table-header-stub>
    <table-body-stub columns=\\"[object Object],[object Object],[object Object]\\" datasource=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\"></table-body-stub>
  </table>
  <pagination-stub></pagination-stub>
</div>"
`;

exports[`columns & !dataSource 1`] = `
"<div>
  <table>
    <table-header-stub columns=\\"[object Object],[object Object],[object Object]\\" sorter=\\"(key, asc) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }\\"></table-header-stub>
    <table-body-stub columns=\\"[object Object],[object Object],[object Object]\\" datasource=\\"\\"></table-body-stub>
  </table>
  <pagination-stub></pagination-stub>
</div>"
`;

exports[`columns & dataSource 1`] = `
"<div>
  <table>
    <table-header-stub columns=\\"[object Object],[object Object],[object Object]\\" sorter=\\"(key, asc) => {
      currentSorter.key = key;
      currentSorter.asc = asc;
    }\\"></table-header-stub>
    <table-body-stub columns=\\"[object Object],[object Object],[object Object]\\" datasource=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\"></table-body-stub>
  </table>
  <pagination-stub></pagination-stub>
</div>"
`;

exports[`mount component 1`] = `
"<div>
      <table>
        <thead>
         <header-row-stub columns=\\"[object Object],[object Object],[object Object]\\" sorter=\\"(key, asc) => {
         currentSorter.key = key;
         currentSorter.asc = asc;
       }\\"></header-row-stub>
       </thead>
       <tbody>
         <body-row-stub columns=\\"[object Object],[object Object],[object Object]\\" record=\\"[object Object]\\"></body-row-stub>
         <body-row-stub columns=\\"[object Object],[object Object],[object Object]\\" record=\\"[object Object]\\"></body-row-stub>
        </tbody>
      </table>
      <pagination-stub current=\\"1\\" disabled=\\"false\\" hideonsinglepage=\\"false\\" pagesize=\\"10\\" showtotal=\\"false\\" total=\\"â€¦
          currentPage.value = page;
        }\\" showpre=\\"false\\" shownext=\\"false\\"></pagination-stub>
    </div>"
`;
